mmdet3d/core/visualizer/show_result.py

```diff
@@ -110,3 +110,70 @@ def show_result(points, gt_bboxes, pred_bboxes, out_dir, filename, show=True):
         pred_bboxes[:, 6] *= -1
         _write_oriented_bbox(pred_bboxes,
                              osp.join(result_path, f'{filename}_pred.ply'))
+
+
+def show_multi_modality_result(img,
+                               gt_bboxes,
+                               pred_bboxes,
+                               proj_mat,
+                               out_dir,
+                               filename,
+                               depth_bbox=False,
+                               img_metas=None,
+                               show=True,
+                               gt_bbox_color=(61, 102, 255),
+                               pred_bbox_color=(241, 101, 72)):
+    """Convert multi-modality detection results into 2D results.
+
+    Project the predicted 3D bbox to 2D image plane and visualize them.
+
+    Args:
+        img (np.ndarray): The numpy array of image in cv2 fashion.
+        gt_bboxes (np.ndarray): Ground truth boxes.
+        pred_bboxes (np.ndarray): Predicted boxes.
+        proj_mat (numpy.array, shape=[4, 4]): The projection matrix
+            according to the camera intrinsic parameters.
+        out_dir (str): Path of output directory
+        filename (str): Filename of the current frame.
+        depth_bbox (bool): Whether we are projecting camera bbox or lidar bbox.
+        img_metas (dict): Used in projecting cameta bbox.
+        show (bool): Visualize the results online. Defaults to True.
+        gt_bbox_color (str or tuple(int)): Color of bbox lines.
+           The tuple of color should be in BGR order. Default: (255, 102, 61)
+        pred_bbox_color (str or tuple(int)): Color of bbox lines.
+           The tuple of color should be in BGR order. Default: (72, 101, 241)
+    """
+    if depth_bbox:
+        from .open3d_vis import draw_depth_bbox3d_on_img as draw_bbox
+    else:
+        from .open3d_vis import draw_lidar_bbox3d_on_img as draw_bbox
```

Comment:
Is it convenient to merge these two functions? It seems that the input/output of them are the same. Using such if/else is a little hard.

Reply:
I do have thought of this before. However, the inner logic of these two functions (projecting 3D bboxes to 2D image plane is very very different), I personally tend to split them as two functions. Is the if/else very dirty in the codebase? I don't know, I thought it is a common practice.

Reply:
OK, then keep it as it is.